### PR TITLE
Warning about improperly configured remote target

### PR DESCRIPTION
**Pull request by rpelisse**
This is a very naive implementation of the fix suggested on the issue #428. Please review and comment!

### CODE REVIEW

File: plugins/connection/podman.py [hunk 1]
```diff
@@ -82,10 +82,17 @@ class Connection(ConnectionBase):
...
+      # compat
+      if _container_id is None and self._play_context.remote_addr is not None:
+        _container_id = self._play_context.remote_addr
+      return _container_id
+
     def __init__(self, play_context, new_stdin, *args, **kwargs):
         super(Connection, self).__init__(play_context, new_stdin, *args, **kwargs)
 
-        self._container_id = self._play_context.remote_addr
+        self._container_id = self._get_container_id
```
Comment (sshnaidm) [inline plugins/connection/podman.py:95]: `self._get_container_id()` ?

Comment (rpelisse) [reply plugins/connection/podman.py]: @sshnaidm thanks!

**Comment by rpelisse** [comment]
```
FATAL: The 1 sanity test(s) listed below (out of 32) failed. See error output above for details.
pep8
```
OK, so there is some [formatting issue](https://peps.python.org/pep-0008/) in my code change. I'll try to fix that asap.

File: plugins/connection/podman.py [hunk 2]
```diff
@@ -82,10 +82,17 @@ class Connection(ConnectionBase):
     # users will start containers and fail to connect to them.
     has_pipelining = False
 
+    def _get_container_id(self):
+        _container_id = self.get_option('remote_addr')
+        # compat
+        if _container_id is None and self._play_context.remote_addr is not None:
+          _container_id = self._play_context.remote_addr
```
Comment (ag0783) [inline plugins/connection/podman.py:89]:
```suggestion
            _container_id = self._play_context.remote_addr
```

**Comment by rpelisse** [comment]
@ag0783 Thanks! I was looking into those formatting issue last week, then I got distracted and never came back to it...

**Comment by sshnaidm** [comment]
This fails with error `The error was: AttributeError: 'Connection' object has no attribute '_load_name'`

**Comment by rpelisse** [comment]
Yep, I saw that. I need to take a look at Python API (as you may have guessed, I'm far from being a python developer) and see what's wrong here.

**Comment by adrianlzt** [comment]
Just adding these two lines fixes the bug
```
--- podman.py.orig      2022-11-08 06:56:09.427326681 +0100
+++ podman.py.new       2022-11-08 07:51:16.936084988 +0100
@@ -23,6 +23,8 @@
         default: inventory_hostname
         vars:
           - name: ansible_host
+          - name: inventory_hostname
+          - name: ansible_podman_host
       remote_user:
         description:
             - User specified via name or UID which is used to execute commands inside the container. If you
```

In fact, just with the ``inventory_hostname`` it is fixed, but I added also de ansible_podman_host to do as in the [docker plugin](https://github.com/bcoca/community.docker/blob/82df7291e22ed3a9b5cdb78c39158d86006b82b3/plugins/connection/docker.py#L32).

I don't fully understand why this works though.

**Comment by m-he** [comment]
> 

it works, thanks a lot!

**Comment by rpelisse** [comment]
Awesome! I'll update the PR today, hopefully, it will finally passes and be ready for merging. Thanks for all the help guys.

**Comment by adrianlzt** [comment]
You can remove the ``self._get_container_id()`` also.

File: plugins/connection/podman.py [hunk 3]
```diff
@@ -82,10 +84,15 @@ class Connection(ConnectionBase):
     # users will start containers and fail to connect to them.
     has_pipelining = False
 
+    def _get_container_id(self):
+        _container_id = self.get_option('remote_addr')
+        if _container_id is None and self._play_context.remote_addr is not None:
+            _container_id = self._play_context.remote_addr
+        return _container_id
```
Comment (sshnaidm) [inline plugins/connection/podman.py:91]: please remove it if you don't use this function

**Review by sshnaidm — CHANGES_REQUESTED** [review]

**Comment by sshnaidm** [comment]
Please sign your commit with `git commit -s` to satisfy DCO job.
You can do `git commit --amend -s` and then push force.

File: plugins/connection/podman.py (hunk 3 above)
Comment (sshnaidm) [reply plugins/connection/podman.py:91]: Didn't you want to remove this function and `_container_id` completely?

Comment (rpelisse) [reply plugins/connection/podman.py:91]: That was the suggestion above, but if I do this there is literally no code changes in the PR (so I suspects it won't fix the problem). 

More importantly, the tests failed complaining about the missing _container_id. So I put both back, to see if the tests passes this time.

Comment (sshnaidm) [reply plugins/connection/podman.py:91]: If I understood @adrianlzt correctly, you don't need to change a code, but just to add these lines:
```
+          - name: inventory_hostname
+          - name: ansible_podman_host
```

**Review by sshnaidm — APPROVED** [review]

**Comment by sshnaidm** [comment]
Probably same thing as in https://github.com/ansible/ansible/issues/77841#issuecomment-1130211123

**Comment by sshnaidm** [comment]
Thanks for your contribution!

**Comment by abalage** [comment]
I ran into this yesterday and I tested the proposed patch manually and it worked for me.
I see the MR is closed now, so all I want to say is thank you.

**Comment by olgeni** [comment]
The fix with "inventory_hostname" will also improve the situation with the jail connection plugin 👀

**Comment by sshnaidm** [comment]
@olgeni you might want to add it here: https://github.com/ansible-collections/community.general/blob/d2094669859fe4b48aa20f1b3ae868b61023b51d/plugins/connection/jail.py#L24-L26

**Comment by olgeni** [comment]
Yup it works ✌️

**Pull request merged**